peer/brontide.go

```diff
@@ -3451,15 +3501,43 @@ func (p *Brontide) handleCloseMsg(msg *closeMsg) {
 
 // HandleLocalCloseChanReqs accepts a *htlcswitch.ChanClose and passes it onto
 // the channelManager goroutine, which will shut down the link and possibly
-// close the channel.
-func (p *Brontide) HandleLocalCloseChanReqs(req *htlcswitch.ChanClose) {
+// close the channel. It takes an additional quit chan as a precaution to avoid
+// potential deadlocks.
+func (p *Brontide) HandleLocalCloseChanReqs(req *htlcswitch.ChanClose,
+	quit chan struct{}) {
+
 	select {
 	case p.localCloseChanReqs <- req:
 		p.log.Info("Local close channel request delivered to " +
 			"peer")
 	case <-p.quit:
-		p.log.Info("Unable to deliver local close channel request " +
-			"to peer")
+		peerLog.Infof("Unable to deliver local close channel request "+
+			"to peer %x", p.PubKey())
+
+	case <-quit:
+		peerLog.Infof("Unable to process local close channel request "+
+			"to peer %x", p.PubKey())
+	}
+}
+
+// HandleCoopReady is called by a ChannelLink to let this subsystem know that
+// the channel is ready to be cooperatively closed. We'll also stop the link.
+// It takes an additional quit chan as a precaution to avoid potential
+// deadlocks.
+func (p *Brontide) HandleCoopReady(chanPoint *wire.OutPoint,
```

Comment:
I named it this since 1) I didn't want the link + peer names to be the same and 2) the link is _notifying_ and the peer is _handling_ (kind of). If it's confusing or easier I can definitely change it tho

Reply:
+1 re `NotifyCoopReady`